src/resources/object.rs

```diff
@@ -177,6 +178,31 @@ impl Object {
         file: Vec<u8>,
         filename: &str,
         mime_type: &str,
+    ) -> crate::Result<Self> {
+        Object::create_with_params(bucket, file, filename, mime_type, None::<&()>).await
+    }
+
+    /// Create a new object.
+    /// Upload a file as that is loaded in memory to google cloud storage, where it will be
+    /// interpreted according to the mime type you specified.
```

Comment:
might want to add a link to the precondition docs here, just so that future folks know which version we're using and what other headers could be used.